### PR TITLE
Removed formslib form, Added manual form, grabbing data

### PR DESCRIPTION
**Pull request by gaetgoler**
#5 fixed

### CODE REVIEW

File: admin/tool/urlpreview/index.php [hunk 1]
```diff
@@ -38,27 +40,24 @@
...
-
-if ($data = $messageform->get_data()) {
-
-    // Fetch the URL and get its metadata.
-    $url = $data->url;
-    // Assuming the unfurl_store class is available.
+$templatedata = [
+    'action' => 'index.php',
+    'formClass' => 'form-class',
+    'urlLabel' => 'URL',
```
Comment (brendanheywood) [inline admin/tool/urlpreview/index.php:50]: You don't need to pass in language string, you can use them directly inside the template like this:

https://moodledev.io/docs/guides/templates#str

I would also hard code the form class, but keep the action as a template variable

File: admin/tool/urlpreview/index.php [hunk 2]
```diff
@@ -38,27 +40,24 @@
...
+    'submitValue' => 'Submit'
+];
+echo $OUTPUT->render_from_template('tool_urlpreview/form', $templatedata);
+
+if ($_SERVER['REQUEST_METHOD'] === 'GET' && isset($_GET['url'])) {
```
Comment (brendanheywood) [inline admin/tool/urlpreview/index.php:55]: this line is redundant, as optional_param does this for you. If the param is not present then $url will be set to ''

File: lib/classes/url/unfurler.php [hunk 3]
```diff
@@ -98,8 +98,8 @@ public function render_unfurl_metadata() {
         // Get the properties of this object as an array.
         $unfurldata = get_object_vars($this);
 
-        // Use the render_from_template method to render your Mustache template.
-        return $OUTPUT->render_from_template('../../../admin/tool/urlpreview/template/metadata', $unfurldata);
+        // Use the render_from_template method to render Mustache template.
+        return $OUTPUT->render_from_template('../../../admin/tool/urlpreview/templates/metadata', $unfurldata);
```
Comment (brendanheywood) [inline lib/classes/url/unfurler.php:102]: can this path be trimmed back to an equivalent but more relative path?

File: admin/tool/urlpreview/index.php (hunk 1 above)
Comment (brendanheywood) [reply admin/tool/urlpreview/index.php:50]: actually I would ditch the form-class and its custom style and just add various pre-existing bootstrap classes 

You can view a similar example here on your local:

https://master.localhost/admin/tool/componentlibrary/docspage.php/moodle/components/search-input/#nav-rendered42

or here in code:

https://github.com/moodle/moodle/blob/master/lib/templates/search_input.mustache

File: admin/tool/urlpreview/templates/form.mustache [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<form action="{{action}}" method="get" class="{{formClass}}">
+    <label for="url">{{urlLabel}}</label>
+    <input type="text" id="url" name="url" required>
```
Comment (brendanheywood) [inline admin/tool/urlpreview/templates/form.mustache:3]: does the url get kept in the input after submission so you can submit a second time?

**Comment by gaetgoler** [comment]
I fixed pointed parts in this branch.
Please let me know if there are further changes are needed!

**Pull request merged**